Tasks/PublishPipelineMetadataV0/publishmetadata.ts

```diff
@@ -0,0 +1,146 @@
+    let requestObjects = [];
+    allVariables.forEach(v => {
+        if (v.name.toLowerCase().startsWith(pipelineMetadataRequestBodyString)) {
+            try {
+                requestObjects.push(JSON.parse(v.value));
+            }
+            catch (error) {
+                tl.debug("Failed to parse metadata variable; Error: " + error);
+            }
+        }
+    });
+
+    return requestObjects;
+}
+
+function constructMetadataRequestBody(requestObject: any): AttestationRequestPayload {
+    if (!requestObject.name) {
+        tl.debug("Not pushing metadata as no name found in request payload");
+        return;
+    }
+
+    const metadata = requestObject.metadata;
+    if (!metadata) {
+        tl.debug("Not pushing metadata as no metadata found");
+        return;
+    }
+
+    if (!metadata.serializedPayload) {
+        tl.debug("Not pushing metadata as no metadata.serializedPayload found");
+        return;
+    }
+
+    let resourceUri: string[] = [];
+    const resourceIds = tl.getVariable("RESOURCE_URIS");
+    if (!requestObject.resourceUris) {
+        if (resourceIds) {
+            const resourceIdArray = resourceIds.split(",");
+            if (resourceIdArray.length == 0) {
+                tl.debug("Not pushing metadata as no resource Ids found");
+                return;
+            }
+            else {
+                resourceUri = resourceIdArray;
+            }
+        }
+        else {
+            tl.loc("Not pushing metadata as no resource Ids found");
+            return;
+        }
+    }
+    else {
+        resourceUri = requestObject.resourceUris;
+    }
+
+    let requestBody: AttestationRequestPayload = {
+        name: requestObject.name,
+        kind: "ATTESTATION",
+        resourceUri: resourceUri,
+        serializedPayload: metadata.serializedPayload
+    };
+
+    if (metadata.description) {
+        requestBody.description = metadata.description;
+    }
+
+    if (metadata.humanReadableName) {
+        requestBody.humanReadableName = metadata.humanReadableName;
+    }
+
+    if (metadata.relatedUrl) {
+        requestBody.relatedUrl = metadata.relatedUrl;
+    }
+
+    return requestBody;
+}
+
+async function run() {
+    try {
+        const metadataObjects = getPipelineMetadataObjects();
+        if (metadataObjects.length == 0) {
+            tl.debug("Not pushing as no metadata found");
+            return;
+        }
+
+        const requestUrl = tl.getVariable("System.TeamFoundationCollectionUri") + tl.getVariable("System.TeamProject") + "/_apis/deployment/attestationdetails?api-version=5.2-preview.1";
+        metadataObjects.forEach((requestPayload: any) => {
+            const requestObject: AttestationRequestPayload = constructMetadataRequestBody(requestPayload);
+            sendRequestToImageStore(JSON.stringify(requestObject), requestUrl).then((result) => {
+                tl.debug("ImageDetailsApiResponse: " + JSON.stringify(result));
```

Comment:
add requestobject to the debug log.

Reply:
We log the request object inside sendRequestToImageStore function itself, so it is not required to log here again